omop_file_validator.py

```diff
@@ -438,13 +522,171 @@ def run_checks(file_path, f):
-    :return: A dictionary of errors found in the file. If there are no errors,
+def run_json_checks(file_path, f):
+    table_name = file_path.stem
+    print(f'Found {file_path.suffix} file %s' % file_path)
+
+    result = {
+        'passed': False,
+        'errors': [],
+        'file_name': file_path.name,
+        'table_name': get_readable_key(table_name)
+    }
+
+    # get the column definitions for a particular OMOP table
+    cdm_table_columns = get_cdm_table_columns(table_name)
+
+    if cdm_table_columns is None:
+        msg = '"%s" is not a valid OMOP table' % table_name
+        print(msg)
+        result['errors'].append(dict(message=msg))
+        return result
+
+    # get column names for this table
+    cdm_column_names = [col['name'] for col in cdm_table_columns]
+
+    if not file_path.exists():
+        print('File does not exist: %s' % file_path)
+        return result
+
+    try:
+        print(f'Parsing JSON Lines file for OMOP table "%s"' % table_name)
+
+        format_errors = check_json_format(f, cdm_column_names)
+
+        for format_error in format_errors:
+            result['errors'].append(
+                dict(message=format_error[0],
+                     actual=format_error[1],
+                     expected=format_error[2]))
+
+        f.seek(0)
+
+        json_list = list(f)
+        row_error_found = False
+
+        for idx, json_str in enumerate(json_list, start=1):
+            if row_error_found:
+                break
+
+            row = pd.read_json(json_str,
+                               nrows=1,
+                               lines=True,
+                               convert_dates=False)
+
+            if is_line_blank(row.iloc[0]):
+                blank_lines_msg = f'File contains blank line on line {idx}.'
+
+                result['errors'].append(dict(message=blank_lines_msg))
+                row_error_found = True
+
+            # check columns if looks good process file
+            if not _check_columns(cdm_column_names, row.columns, result):
+                row_error_found = True
+
+            #search for scientific notation
+            int_columns = [
+                col['name'] for col in cdm_table_columns
+                if col['type'] == 'integer'
+            ]
+
+            sci_not_line = has_scientific_notation_error(row, int_columns)
+            for col, (value, line_num) in sci_not_line.items():
+                sci_not_error_msg = dict(message=(
+                    f"Scientific notation value '{value}' was found on line {idx}. "
+                    "Scientific notation is not allowed for integer fields."),
+                                         column_name=col)
+                result['errors'].append(sci_not_error_msg)
+                row_error_found = True
+
+            for meta_item in cdm_table_columns:
+                meta_column_name = meta_item['name']
+                meta_column_required = meta_item['mode'] == 'required'
```

Comment:
nit: Not an issue here, but would suggest using .lower() for comparisons like this.
```suggestion
                meta_column_required = meta_item['mode'].lower() == 'required'
```